### DISCUSSION TITLE
how to change color of sash (seperator ) in panedwindow.

### DISCUSSION THREAD

**Discussion by its-me-abi** [israel-dryer/ttkbootstrap#604] [Q&A] [answered]
to change sash color of ttk.panedwindow I tried everything, like changing bordercolor, borderwidth, sashcolor, etc., but nothing worked. When I added bootstyle as an argument to the PanedWindow constructor, it allowed color changes, but only a limited set of predefined colors.

I tried to debug it by reading the source code and found that the create_panedwindow_style function is responsible for changing the sash color, 

I edited this function of this module and added sashcolor to make it work, but that's not a real solution. Has anyone found other options in the API to change the sash color?

**Comment by israel-dryer** [comment, marked as answer]
The `bootstyle` argument only maps to the theme's predefined semantic colors (primary/success/info/…), which is why you saw a limited set. For an **arbitrary** sash color, set the `background` of the panedwindow style directly — that's the same option ttkbootstrap uses internally to color the sash:

```python
import ttkbootstrap as ttk

app = ttk.Window()
pw = ttk.Panedwindow(app, orient="horizontal")   # note: Panedwindow
pw.add(ttk.Frame(pw)); pw.add(ttk.Frame(pw))
pw.pack(fill="both", expand=True)

# horizontal panedwindows use Horizontal.TPanedwindow; vertical use Vertical.TPanedwindow
app.style.configure("Horizontal.TPanedwindow", background="#e83e8c")   # AFTER the widget
```

Two things to know:

1. Configure it **after** the widget exists — ttkbootstrap builds the panedwindow style lazily, so configuring earlier gets overwritten when the style is first built.
2. A `theme_use()` switch rebuilds the style and resets the color. If you change themes at runtime, re-apply on `<<ThemeChanged>>`:

```python
def _apply(_=None):
    app.style.configure("Horizontal.TPanedwindow", background="#e83e8c")
_apply()
app.bind("<<ThemeChanged>>", _apply, "+")
```

Your instinct that editing `create_panedwindow_style` "isn't a real solution" is right — the lack of a documented, theme-durable way to set sash color/thickness is tracked for 2.0 in #1161.
